tests/Middleware/BufferTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace React\Tests\Http\Middleware;
+
+use Psr\Http\Message\ServerRequestInterface;
+use React\Http\Middleware\Buffer;
+use React\Http\Response;
+use React\Http\ServerRequest;
+use React\Tests\Http\TestCase;
+use RingCentral\Psr7\BufferStream;
+
+final class BufferTest extends TestCase
+{
+    public function testBuffer()
+    {
+        $size = 1024;
+        $body = str_repeat('x', $size);
+        $stream = new BufferStream($size);
+        $stream->write($body);
+        $serverRequest = new ServerRequest(
+            'GET',
+            'https://example.com/',
+            array(),
+            $stream
+        );
+
+        $exposedRequest = null;
+        $buffer = new Buffer();
+        $buffer(
+            $serverRequest,
+            function (ServerRequestInterface $request) use (&$exposedRequest) {
+                $exposedRequest = $request;
+            }
+        );
+
+        $this->assertSame($body, $exposedRequest->getBody()->getContents());
+    }
+
+    public function testToLargeBody()
+    {
+        $size = $this->iniMaxPostSize() + 1;
+        $stream = new BufferStream($size);
```

Comment:
Not sure what's supposed to be tested here? Afaict this should probably use a fixed limit instead?